Updates to Java Starter Project

Nettogrof opened the pull request:
As suggested in the [feedback discussion](https://github.com/BattlesnakeOfficial/feedback/discussions/114), 

1. I added a  "avoid neck function" to avoid to collide with its neck.
2.  Added and update comments to be similar to comments in others starter-snake project.
3.  Added tests for the "avec neck function" 

I also update the ReadMe file, to be more similar to other project readme, and remove any "Slack" reference.

Nettogrof committed:
Fixing Head and Tail Customizations issue #8

Nettogrof committed:
Change Repl.it -> replit.com

Nettogrof committed:
Fix typo

Nettogrof committed:
Merge branch 'BattlesnakeOfficial:master' into master

Nettogrof committed:
update ReadMe.md - remove slack info

Removing the slack server info

snyk-bot committed:
fix: pom.xml to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1009829
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1048302
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1052449
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1052450
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1054588
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056414
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056416
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056417
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056418
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056419
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056420
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056421
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056424
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056425
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056426
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1056427
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-1061931
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-570625
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-572300
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-572314
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-572316
- https://snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-608664
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-174560
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-32381
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-32383
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-32384
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-32385
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-460763
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-461008
- https://snyk.io/vuln/SNYK-JAVA-ORGECLIPSEJETTY-461009

Nettogrof committed:
Merge pull request #1 from Nettogrof/snyk-fix-f3aee04169810131f010d4e970b33835

[Snyk] Fix for 30 vulnerabilities

snyk-bot committed:
fix: upgrade com.fasterxml.jackson.core:jackson-databind from 2.9.10.8 to 2.12.3

Snyk has created this PR to upgrade com.fasterxml.jackson.core:jackson-databind from 2.9.10.8 to 2.12.3.

See this package in Maven Repository:
https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind/

See this project in Snyk:
https://app.snyk.io/org/nettogrof/project/c0596349-eb7f-4aaa-8b73-72af05c52a81?utm_source=github&utm_medium=upgrade-pr

snyk-bot committed:
fix: upgrade org.slf4j:slf4j-simple from 1.7.21 to 1.7.30

Snyk has created this PR to upgrade org.slf4j:slf4j-simple from 1.7.21 to 1.7.30.

See this package in Maven Repository:
https://mvnrepository.com/artifact/org.slf4j/slf4j-simple/

See this project in Snyk:
https://app.snyk.io/org/nettogrof/project/c0596349-eb7f-4aaa-8b73-72af05c52a81?utm_source=github&utm_medium=upgrade-pr

snyk-bot committed:
fix: upgrade com.sparkjava:spark-core from 2.9.1 to 2.9.3

Snyk has created this PR to upgrade com.sparkjava:spark-core from 2.9.1 to 2.9.3.

See this package in Maven Repository:
https://mvnrepository.com/artifact/com.sparkjava/spark-core/

See this project in Snyk:
https://app.snyk.io/org/nettogrof/project/c0596349-eb7f-4aaa-8b73-72af05c52a81?utm_source=github&utm_medium=upgrade-pr

Nettogrof committed:
Merge pull request #4 from Nettogrof/snyk-upgrade-ee5439e7776fc6e899b2f0c613c54ea8

[Snyk] Upgrade com.sparkjava:spark-core from 2.9.1 to 2.9.3

Nettogrof committed:
Merge pull request #3 from Nettogrof/snyk-upgrade-c9e2402500ff5fed9f7e3d2f498fb422

[Snyk] Upgrade org.slf4j:slf4j-simple from 1.7.21 to 1.7.30

Nettogrof committed:
Merge pull request #2 from Nettogrof/snyk-upgrade-c9dbed149dbfee6f7627a39a03897dee

[Snyk] Upgrade com.fasterxml.jackson.core:jackson-databind from 2.9.10.8 to 2.12.3

snyk-bot committed:
fix: upgrade org.slf4j:slf4j-simple from 1.7.30 to 1.7.31

Snyk has created this PR to upgrade org.slf4j:slf4j-simple from 1.7.30 to 1.7.31.

See this package in Maven Repository:
https://mvnrepository.com/artifact/org.slf4j/slf4j-simple/

See this project in Snyk:
https://app.snyk.io/org/nettogrof/project/c0596349-eb7f-4aaa-8b73-72af05c52a81?utm_source=github&utm_medium=upgrade-pr

snyk-bot committed:
fix: upgrade com.fasterxml.jackson.core:jackson-databind from 2.12.3 to 2.12.4

Snyk has created this PR to upgrade com.fasterxml.jackson.core:jackson-databind from 2.12.3 to 2.12.4.

See this package in Maven Repository:
https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind/

See this project in Snyk:
https://app.snyk.io/org/nettogrof/project/c0596349-eb7f-4aaa-8b73-72af05c52a81?utm_source=github&utm_medium=upgrade-pr

Nettogrof committed:
Merge branch 'BattlesnakeOfficial:master' into master

Nettogrof committed:
Merge pull request #6 from Nettogrof/snyk-upgrade-a2693252ca1b914f29c222b3ee0dff8b

[Snyk] Upgrade com.fasterxml.jackson.core:jackson-databind from 2.12.3 to 2.12.4

Nettogrof committed:
Merge pull request #5 from Nettogrof/snyk-upgrade-e71e50f8c38529ac3a3c0a0d943b19f2

[Snyk] Upgrade org.slf4j:slf4j-simple from 1.7.30 to 1.7.31

Nettogrof committed:
Merge branch 'BattlesnakeOfficial:master' into master

Nettogrof committed:
AvoidNeck + comments

Added a function to avoidMyNeck to move selection. And added TODO comments to let new developers knows next steps to do.

Nettogrof committed:
Fix errors in the indexTest, and provide a real payload for the moveTest

Nettogrof committed:
Fix maven warning and remomve duplicate plugin import

Nettogrof committed:
Add proper comments

Nettogrof committed:
Adding avoidNeck test cases

Nettogrof committed:
Clean-upo the readMe to be more like readme files from official
starter-snake project

Nettogrof committed:
Few minor changes in comments for clarity and to respect Java standard

bvanvugt commented:
Thanks @Nettogrof! We'll take a look this week, as I am not qualified :)

aurorawalker commented on README.md:
```diff
@@ -1,246 +1,199 @@
+
+3. You should see your Repl being initialized - this might take a few moments to complete.
+
+4. Once your Repl is ready to run, click `Run â–¶ï¸�` at the top of the screen. You should see maven (and any other dependencies) being installed. Once installation is complete, your Battlesnake server will start and you should see the following:
```
Whoop - looks like the IDE or similar did something odd with the 'Run >' part of this.

aurorawalker commented on src/main/java/com/battlesnake/starter/Snake.java:
```diff
@@ -1,180 +1,232 @@
+		 * For a full example of 'json', see
+		 * https://docs.battlesnake.com/references/api/sample-move-request
+		 *
+		 * @param moveRequest JsonNoe of all Game Board data as received from the
```
Small typo - JsonNoe  - > JsonNode

aurorawalker commented on src/main/java/com/battlesnake/starter/Snake.java:
```diff
@@ -1,180 +1,232 @@
+			// TODO Using information from 'data', don't let your Battlesnake pick a move
+			// that would hit its own body
+
+			// TODO: Using information from 'data', don't let your Battlesnake pick a move
```
I think the comment here (and one below on line 179) is a bit too literally copied from the Python starter project comments. I think instead of saying " Using information from 'data'..." it would be better to say "Using information from 'moveRequest'..." as that is the name of the object in this Java starter project.

Nettogrof replied:
I plead guilty of  "copy-pasta" ;)

aurorawalker replied:
Literally happens to all of us. That's why I love code review so much - means I've got a buddy checking my work for when I inevitably copy-pasta wrong. :D Code reviewers make up for most standard human errors.

aurorawalker reviewed (changes requested):
Thank you for your thoroughness! Great PR. I've done a visual review, spotted some minor typos. Going to publish those comments now, but next step for me will be running the complete PR and taking it for a spin. :D

Nettogrof committed:
Fix based on comments

1. Fix typo JsonNoe->JsonNode
2. Change 'data' to 'moverequest'
3. Fix the 'Run >'

Nettogrof commented:
All should be fixed now

aurorawalker commented:
@Nettogrof 
Functionally looks good! Successfully was able to create and run a Replit, as well as run the unit test suite. Great job, and thank you for this!

Sorry to be a nitpicking-Nancy, but I think there is one last thing I'd like to follow up on. The pull request shows an edit to more lines that there should be, so I pulled the repo down and see that the files you modified also had the line ending changed to CR/LF (carriage return and line feed) while the rest of the repository is just LF (line feed).
CR/LF is what you see on Windows, where LF is what you'll see on unix systems.

The other files in the repo have LF line endings, and I'd like to stay consistent on that and keep all files as LF. If you use Visual Studio Code as your editor, there is a handy toggle button between LF and CR/LF in the bottom-right-hand corner.
<img width="332" alt="vscode_lf" src="https://user-images.githubusercontent.com/663009/136477413-7cd09115-2717-4dd5-a7a8-38e4b7075896.png">

Nettogrof committed:
Change newLine from CRLF to LF

Nettogrof committed:
Changing Tabs to space identation

aurorawalker reviewed (approved):
You are a champ! Thanks for addressing the nitpicks. Great work.
